glib/src/functions.rs

```diff
@@ -24,15 +24,15 @@ use crate::{Error, Pid, SpawnFlags};
 #[cfg_attr(docsrs, doc(cfg(all(feature = "v2_58", not(windows)))))]
 #[allow(clippy::too_many_arguments)]
 #[doc(alias = "g_spawn_async_with_fds")]
-pub fn spawn_async_with_fds<P: AsRef<std::path::Path>, T: AsRawFd, U: AsRawFd, V: AsRawFd>(
+pub fn spawn_async_with_fds<P: AsRef<std::path::Path>>(
     working_directory: P,
     argv: &[&str],
     envp: &[&str],
     flags: SpawnFlags,
     child_setup: Option<Box_<dyn FnOnce() + 'static>>,
-    stdin_fd: T,
-    stdout_fd: U,
-    stderr_fd: V,
+    stdin_fd: Option<impl AsFd>,
+    stdout_fd: Option<impl AsFd>,
+    stderr_fd: Option<impl AsFd>,
```

Comment:
`Option<impl ...>` is usually a bad idea. Just try calling this function with a `None` :)

I think here and in the other APIs using an `Option<impl ...>` currently you'll have to add a builder pattern instead